book/src/development/method_checking.md

```diff
@@ -0,0 +1,93 @@
+# Method Checking
+
+In some scenarios we might want to check for methods when developing
+a lint. There are two kinds of questions that we might be curious about:
+
+- Invocation: Does an expression call a specific method?
+- Definition: Does the type `Ty` of an expression define a method?
```

Comment:
This says something different than what is then documented below. Below is documented on how to check an `impl` block. This is pretty trivial and doesn't really need documentation IMO. So I would leave this out.

I think usually, we don't need to check if a `Ty` implements a certain method, but rather if a `Ty` implements a trait. But there's a different chapter for this.

Let me know if you disagree.

Reply:
I think we can implement both checking if a method call calls a specific method **and** if the type implements a trait in the same example (The comments explain it). And center the other one about checking an `impl` instead of a `Ty` (Changing the wording, keeping the example)